backtracking/minimax.py

```diff
@@ -1,4 +1,7 @@
+from __future__ import annotations
 import math
+from typing import List
+
```

Comment:
```suggestion
from __future__ import annotations

import math
```
With the annotations import, we no longer need to from typing import List but we need to change all instances of List --> list in the code below.  See the changes made in #2464